tests/Paket.Tests/Lockfile/GeneratorSpecs.fs

```diff
@@ -177,22 +177,36 @@ let ``should generate lock file with disabled content for packages``() =
     |> LockFileSerializer.serializePackages cfg.Groups.[Constants.MainDependencyGroup].Options
     |> shouldEqual (normalizeLineEndings expected)
 
-let expectedWithGitHub = """GITHUB
+[<Test>]
+let ``should generate lock file for github source files``() =
+    let expectedWithGitHub = "GITHUB
   remote: owner/project1
-    "folder/file 2.fs" (commit1)
+    \"folder/file 2.fs\" (commit1)
     folder/file.fs (master)
     folder/file1.fs (commit1)
+    folder/file3.fs (feature/branch)
+    folder/file4.fs (feature/branch)
+    folder/file5.fs (feature/branch)
+    folder/file6.fs (feature/branch)
+    folder/file7.fs (feature/branch)
   remote: owner/project2
     folder/file.fs (commit2)
-    folder/file3.fs (commit3) githubAuth"""
-    
-[<Test>]
-let ``should generate lock file for source files``() = 
-    let config = """github "owner:project1:master" "folder/file.fs"
-github "owner/project1:commit1" "folder/file1.fs"
-github "owner/project1:commit1" "folder/file 2.fs"
-github "owner:project2:commit2" "folder/file.fs"
-github "owner:project2:commit3" "folder/file3.fs" githubAuth """ 
+    folder/file3.fs (commit3) githubAuth
+  remote: owner/project3
+    FULLPROJECT (master)"
+
+
+    let config = "github \"owner:project1:master\" \"folder/file.fs\"
+github \"owner:project1:feature/branch\" \"folder/file3.fs\"
+github \"owner/project1:feature:branch\" \"folder/file4.fs\"
```

Comment:
Should we really allow that?

Reply:
Don't see why not...you can create a branch called that I believe?

Reply:
pretty common with branching strategies like [gitflow](https://datasift.github.io/gitflow/IntroducingGitFlow.html)

Reply:
@matthid updated to no longer allow this.
Also added a range of other test cases :)